galsim/phase_psf.py

```diff
@@ -1085,6 +1094,14 @@ def _finalize(self, flux, suppress_warning):
         self.img *= (flux / (self.img.sum() * self.scale**2))
         self.img = galsim.ImageD(self.img.astype(np.float64), scale=self.scale)
 
+        if self.max_size is not None and self.max_size < self.img.scale*self.img.array.shape[0]:
+            center = self.img.center()
+            bds = galsim.BoundsI(center)
+            half_max_size = int(np.ceil(0.5*self.max_size/self.img.scale))
+            bds += center + galsim.PositionI(-half_max_size, -half_max_size)
+            bds += center + galsim.PositionI(half_max_size, half_max_size)
+            self.img = self.img[bds]
```

Comment:
>But if you think this is invalid, I'll take your word for it, since you understand this stuff much better than I do.

I was about 80% confident before, but I just ran a test and now I'm up to 99%.  Here's a test script using the deprecated OpticalPSF class:

```
import galsim
import numpy as np
import matplotlib.pyplot as plt
psf = galsim.optics.OpticalPSF(lam_over_diam=1, nstruts=3)
print psf._optimage # galsim.Image(bounds=galsim.BoundsI(1,384,1,384), scale=0.333333333333)
print 384/3. # 128 (arcsec) is the internal image size
# Plot the inner 64 arcsec by 64 arcsec part:
plt.imshow(np.log10(psf._optimage.array[96:-96, 96:-96]), vmin=-10, vmax=0)
plt.colorbar()
plt.show()
# Now use max_size = 64, i.e., make the inner 50% of the image.
psf2 = galsim.optics.OpticalPSF(lam_over_diam=1, nstruts=3, max_size=64)
print psf2._optimage # galsim.Image(bounds=galsim.BoundsI(1,192,1,192), scale=0.333333333333)
# And show the entire second image, which is already clipped to 64 arcsec by max_size.
plt.imshow(np.log10(psf2._optimage.array), vmin=-10, vmax=0)
plt.colorbar()
plt.show()
```

Image without max_size:

![no_max_size](https://cloud.githubusercontent.com/assets/3650485/16062725/f6471102-3249-11e6-996f-4a9efc6ab053.png)

Image with max_size:

![max_size](https://cloud.githubusercontent.com/assets/3650485/16062735/fd5235a8-3249-11e6-9000-4a58669969b6.png)

You can see the additional aliases as the non-radial diffraction spikes in the second image, which are really radial spikes originating from the neighboring "copies" of the PSF above and below that got folded over into the "fundamental" image.  (Not sure if I'm using the right Fourier lingo here, but hopefully this is clear!)  It's definitely a small effect (~10^-4), but I think it's better handled through pad_factor than max_size.

So I went ahead and removed the max_size kwarg completely from PhaseScreenPSF, and deprecated it in the new version of OpticalPSF.